service/history/api/update_workflow_util.go

```diff
@@ -96,6 +96,9 @@ func UpdateWorkflowWithNew(
 		if err != nil {
 			return err
 		}
+		// if current workflow gets an update with a higher version than
+		// the version of the new workflow, it means failover has happened,
+		// and the new workflow creation should be failed
```

Comment:
NOTE for reviewers: I added some explanation on why continue to use LastWriteVersion is safe at multiple places. 
Those will be removed before landing.